pages/api/paybutton/transactions/[id].ts

```diff
@@ -0,0 +1,23 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { fetchTransactionsByPaybuttonId } from 'services/transactionService'
+
+export default async ( req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+  if (req.method === 'GET') {
+    const paybuttonId = req.query.id as string;
+    try {
+      const transactions = await fetchTransactionsByPaybuttonId(paybuttonId);
+      
+      res
+        .status(200)
+        .json({ transactions });
+    } catch (err: any) {
+      res
+        .status(500)
+        .json({ statusCode: 500, message: err.message })
+    }
+  } else {
+    res
+      .status(405)
```

Comment:
Where is this gonna be thrown?

Also, missing case for NO_TRANSACTION_FOUND_404

Reply:
the " method not allowed " error is something present in almost every /api endpoint 
added case for 404

Reply:
Looks like a good idea indeed, but since we should be using this for endpoints from now on, you can create a variable in RESPONSE_MESSAGES like it's done everywhere else.